src/components/TodoComponent/TodoComponent.tsx

```diff
@@ -0,0 +1,117 @@
+import classNames from 'classnames';
+import React, { useEffect, useRef, useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo,
+  isLoading: boolean,
+  onDelete: (id: number) => void
+  onUpdateTodo: (id: number, todo: object) => void
```

Comment:
Instead of using `object` as the type for `todo`, it's better to use a more specific type. Consider using `Partial<Todo>` as a type for the updated todo to provide better type-checking.